### PR TITLE
#3262 - Ministry Application Tracker UI

### PR DESCRIPTION
**Pull request by dheepak-aot**
# Ministry Application Tracker UI

## Component centralization and Refactoring

- [x] The application status tracker component moved to `components/common`
- [x] Stopped passing `applicationStatus` as property to the `ApplicationProgressBar` component. Only `applicationId` is passed and optionally `isApplicationActionsAllowed` to disable the actions from the application tracker.
- [x] Removed the deprecated `MenuModel` from side bar component and replaced it with `MenuItemModel`

## Manual Testing

- [x] Draft Application
image

- [x] Cancelled Application
image

- [x] InProgress(Exception)
image

- [x] InProgress(Waiting Partner Input)
image

- [x] Completed Application With Original Assessment
image

- [x] Completed Application with Application Offering Change requested
image

### CODE REVIEW

File: sources/packages/web/src/components/common/applicationTracker/Draft.vue
Comment (andrewsignori-aot) [inline sources/packages/web/src/components/common/applicationTracker/Draft.vue:25]: Minor, but should it be `areApplicationActionsAllowed`?

File: sources/packages/web/src/constants/routes/RouteConstants.ts
Comment (andrewsignori-aot) [inline sources/packages/web/src/constants/routes/RouteConstants.ts:153]: It may be generic. Should it be, for instance, `APPLICATION_STATUS_TRACKER`?

File: sources/packages/web/src/types/AppRoutes.ts
Comment (andrewsignori-aot) [inline sources/packages/web/src/types/AppRoutes.ts:123]: Same here https://github.com/bcgov/SIMS/pull/3756/files#r1785269842

**Review by andrewsignori-aot — COMMENTED** [review]
Great work, it is looking so good on the Ministry side. Only minor, non blocker, comments.

**Comment by lewischen-aot** [comment]
PR is looking great. Waiting for the primary reviewer's approval.

File: sources/packages/web/src/constants/routes/RouteConstants.ts
Comment (dheepak-aot) [reply sources/packages/web/src/constants/routes/RouteConstants.ts]: Sure, Just kept the title of side nav button as `Application Status` everything else is modified to have `tracker` in the end.

**Review by andrewsignori-aot — APPROVED** [review]
Thanks for making the changes, Looks good 👍

**Review by lewischen-aot — APPROVED** [review]
LGTM 👍 Thanks for making the changes.

**Pull request merged**
